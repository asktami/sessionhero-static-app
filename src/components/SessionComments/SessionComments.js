import React, { Component } from 'react';
import AppContext from '../../contexts/AppContext';
import { Link } from 'react-router-dom';
import { SessionStarRating } from '../../components/SessionStarRating/SessionStarRating';

export default class SessionContents extends Component {
	static contextType = AppContext;

	render() {
		const { session, comments, deleteComment } = this.context;

		return (
			<ul className="comment-list">
				{(comments || []).map(comment => (
					<li key={comment.id} className="comment-item">
						<div className="comment-text">
							{comment.text}
							<br />

							<div className="flex-row comment-footer ">
								<div>
									<SessionStarRating rating={comment.rating} />
									<br />
									<span className="comment-user sponsor">
										{comment.userId === 1
											? 'LoggedIn User Created This Comment'
											: 'OtherFirst OtherLast'}
									</span>
								</div>
								{/* TBD */}
								{comment.userId === 1 ? (
									<div className="flex-row comment-btns">
										<div>
											<Link
												to={`/comments/${comment.id}?session=${session.name}`}
											>
												<button className="btn-edit-comment">Edit</button>
											</Link>
										</div>
										<div>
											<button
												className="btn-delete-comment"
												onClick={() => deleteComment(comment.id)}
											>
												Delete
											</button>
										</div>
									</div>
								) : null}
							</div>
						</div>
					</li>
				))}
			</ul>
		);
	}
}
