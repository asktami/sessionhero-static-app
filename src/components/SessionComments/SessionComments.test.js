import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SessionComments from './SessionComments';

describe(`SessionComments component`, () => {
	const props = {
		match: {
			params: {
				sessionId: 'BUS04'
			}
		},
		comments: [
			{
				id: 1,
				text: 'This thing is amazing.',
				rating: 4,
				date_created: '2019-10-31 23:14:12.649275',
				sessionId: 'BUS04',
				userId: 1
			},
			{
				id: 2,
				text: 'Put a bird on it!',
				rating: 4,
				date_created: '2019-10-31 23:14:12.649275',
				sessionId: 'BUS04',
				userId: 1
			}
		]
	};

	it('renders a SessionComments by default', () => {
		const wrapper = shallow(<SessionComments {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the SessionComments given props', () => {
		const wrapper = shallow(<SessionComments {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
