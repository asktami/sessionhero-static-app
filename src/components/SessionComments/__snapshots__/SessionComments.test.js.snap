// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionComments component renders a SessionComments by default 1`] = `
<ul
  className="comment-list"
/>
`;

exports[`SessionComments component renders the SessionComments given props 1`] = `
<ul
  className="comment-list"
/>
`;
